devices/Atgm336h/README.md

```diff
@@ -0,0 +1,58 @@
+# GPS Module
+ATGM336H is a highly sensitive GPS module that provides high accuracy navigation and positioning features.
+
+## Documentation
+
+`ATGM336H` GPS module relies on the `NMEA0183` standard for communication. This standard is widely used for the interfacing and decoding of navigational data from the GPS module.
+
+For more understanding and better implementation, please refer to the official [NMEA0183 documentation](https://www.tronico.fi/OH6NT/docs/NMEA0183.pdf).
+
+## Connections
+
+There are only 4 wires required:
+
+| ATGM336H | MCU Header |
+|----------|------------|
+| VCC      | 3.3V       |
+| GND      | GND        |
+| TX       | RX         |
+| RX       | TX         |
+
+## Usage
+
+Takes up to 30s to get fix. There is an internal diode on module that indicates if the fix is acquired.
+
+```csharp
+using Iot.Device.Atgm336h;
+using System;
+using System.Threading;
+using nanoFramework.Hardware.Esp32;
+
+var rxPin = 21;
+var txPin = 19;
+var comPort = "COM2";
+Configuration.SetPinFunction(rxPin, DeviceFunction.COM2_RX);
+Configuration.SetPinFunction(txPin, DeviceFunction.COM2_TX);
+var gpsModule = new Atgm336h(comPort);
+gpsModule.LocationChanged += GpsModuleOnLocationChanged;
+gpsModule.FixChanged += GpsModuleOnFixChanged;
+gpsModule.ModeChanged += GpsModuleOnModeChanged;
+gpsModule.Start();
+
+Thread.Sleep(Timeout.Infinite);
+
+void GpsModuleOnFixChanged(Fix fix)
```

Comment:
and another question for a longer discussion: why not having an abstract class as well in a simple nuget with the generic parser that already implement functions like this one?
All GPS have something similar. In short, we would have a basic parser, the base class with the event, the fix, the modes, all this. And then the ATGM336H would just have to implement what is specific.

Reply:
Abstract class or maybe static helper? Like WeatherHelper class (maybe it should be named GpsHelper)?

Reply:
I think all GPS are quire similar. And the even they can raise are similar as well. So having an abstract class at minima, with possibly static helpers can be a great combination.
I guess the idea is to keep it nano compatible, simple and efficient. The .NET IoT repo code is too complex for nano. But it has the advantage of containing all the methos. Also, it has tests and other elements that can be reused.

Reply:
done